app/src/main/java/com/example/peterchu/watplanner/home/HomeFragment.java

```diff
@@ -200,16 +204,24 @@ protected boolean openDetailView(Integer courseId) {
         return true;
     }
 
-    public void setCourseSchedule(List<CourseComponent> courseSchedule) {
+    public void setCourseSchedule(List<List<CourseComponent>> courseSchedule) {
         mCourseSchedule = courseSchedule;
         // The week view has infinite scrolling horizontally. We have to provide the events of a
         // month every time the month changes on the week view.
         weekView.setMonthChangeListener(new MonthLoader.MonthChangeListener() {
             @Override
             public List<? extends WeekViewEvent> onMonthChange(int newYear, int newMonth) {
                 List<WeekViewEvent> events = new ArrayList<>();
-                for (CourseComponent c : mCourseSchedule) {
-                    events.addAll(c.toWeekViewEvents(newMonth));
+                // Color each section and type a different color and add to view
+                Random rnd = new Random();
+                for (List<CourseComponent> components : mCourseSchedule) {
+                    int color = Color.argb(255, rnd.nextInt(256), rnd.nextInt(256), rnd.nextInt(256));
```

Comment:
I agree @PeterChu2 , the other option I considered was having a color rotation array in an arrays.xml and then just rotating through (ensuring my have a high enough count).

@Mrez95 are we dividing up R, G, and B? e.g. 51/51/51, 102/102/102? That will result in different shades of like grey and black. Might be too similar which relates back to @PeterChu2's concern.